### ISSUE TITLE
Can spawn in nothingness in small worlds

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#8840]
## Description
If you play in a small world (small value of `mapgen_limit`), it's possible you spawn in nothingness, i.e. outside the map bounds.

## How to reproduce

* Set `mapgen_limit` to 100
* Create a new world with v7 mapgen
* Spawn

## Version
5.0.1

## My guess

I strongly suspect this is related to the new spawn code, which apparently isn't smart enough to “see” the map boundaries, so it puts the player

**Comment by paramat** [comment]
5.0.1 but which commit? I assume after 37923920a07f06d201662d8f1f4e5821efcc7b09 ?

**Comment by Wuzzy2** [comment]
Sigh. I have posted it to `minetest/minetest_game` for a reason. It's because I reproduced it in Minetest Game only. I did not reproduce it in other games.

And with “new spawn code” I meant the spawning mod of Minetest Game. I didn't check what happens if I remove this mod.

**Comment by paramat** [comment]
Aha, however it may be an engine issue, i'll investigate.

**Comment by paramat** [comment]
![screenshot_20190822_004533](https://user-images.githubusercontent.com/3686677/63475785-3d96f880-c476-11e9-89d5-3e7dd6f96121.png)

Yes confirmed when MTG spawn mod is enabled.

**Comment by Wuzzy2** [comment]
Off topic: What does “5.0.1 but which commit mean”? Aren't version numbers uniquely mapping to a commit anymore? o_O

On topic: I spawn very often, maybe even always, outside of map bounds in MTG. The coordinates are not always the same, but the last time it was X=-128, Y=11, Z=64. This never happens in games that are not MTG.

---

I strongly suspect the spawning mod of MTG to be the culprit because it messes with player pos.

**Comment by Wuzzy2** [comment]
Oh, this might be related or maybe not: I noticed that `minetest.get_biome_data` always returns valid biomes, even if the position is out of map bounds. Intentional?

**Comment by paramat** [comment]
Sorry, forgot we are 5.1.0-dev. So you mean 5.0.1 stable.

The XZ coords being multiples of 64 shows it is caused by the 'spawn' mod.

**Comment by paramat** [comment]
Yes expected, as heat / humidity noises still function for coordinates outside mapgen limit.
It's correct data because it would be correct if the world generated there.

**Comment by paramat** [comment]
Looks like the problem is caused by the MTG spawn mod not checking positions for being outside 'mapgen_limit' as you suspected, so is a MTG issue.
